ts/nni_manager/common/utils.ts

```diff
@@ -449,8 +450,49 @@ function withLockSync(func: Function, filePath: string, lockOpts: {[key: string]
     return result;
 }
 
+async function isPortOpen(host: string, port: number): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+        try{
+            const stream = net.createConnection(port, host);
+            const id = setTimeout(function () {
+                stream.destroy();
+                resolve(false);
+            }, 1000);
+
+            stream.on('connect', function () {
+                clearTimeout(id);
+                stream.destroy();
+                resolve(true);
+            });
+
+            stream.on('error', function () {
+                clearTimeout(id);
+                stream.destroy();
+                resolve(false);
+            });
+        } catch (error) {
+            reject(error);
+        }
+    });
+}
+
+async function getFreePort(host: string, start: number, end: number): Promise<number> {
+    const deferred = new Deferred<number>();
```

Comment:
Seems `await` is enough for this function. No need to use `Deferred`.

Reply:
fix it